src/main/java/actions/LocomotiveAction.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package actions;
+
+import components.LocomotiveComponent;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.entitySystem.systems.UpdateSubscriberSystem;
+import org.terasology.logic.common.ActivateEvent;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.minecarts.components.RailVehicleComponent;
+import org.terasology.minecarts.controllers.CartMotionSystem;
+import org.terasology.registry.In;
+import org.terasology.segmentedpaths.components.PathFollowerComponent;
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class LocomotiveAction extends BaseComponentSystem implements UpdateSubscriberSystem {
```

Comment:
Some minor class level javadoc would be nice. 👍 

Also not necessarily a thing for this PR but I'm beginning to wonder if we'll cause confusion with System classes with "Action" in them - there are plenty already, but `Action` is also an interface related to the Behavior Tree framework. Maybe we can pick some better terminology specific to Systems that handle "actions" of some sort. I suspect that naming just lives on via copy pasting examples.